plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,629 @@
+                                mso_values[(protocol_name, "authEnabled")] = protocol_dict.get("authentication")
+                            mso_values[(protocol_name, "keyID")] = protocol_dict.get("key_id")
+
+                            if protocol_dict.get("key") is not None:
+                                if match.details.get(protocol_name, {}).get("key") is not None:
+                                    mso_values[(protocol_name, "key", "value")] = protocol_dict.get("key")
+                                else:
+                                    mso_values[(protocol_name, "key")] = dict(value=protocol_dict.get("key"))
+                        else:
+                            if protocol_name == "ospf" and match.details.get(protocol_name, {}).get("authType"):
+                                mso_values_remove.extend([(protocol_name, "key"), (protocol_name, "keyID"), (protocol_name, "authType")])
+                            elif match.details.get(protocol_name, {}).get("authEnabled"):
+                                mso_values_remove.extend([(protocol_name, "key"), (protocol_name, "keyID"), (protocol_name, "authEnabled")])
+
+            append_update_ops_data(ops, match.details, interface_group_policy_path, mso_values, mso_values_remove)
+            mso.sanitize(match.details, collate=True)
+        else:
+            update_mso_values_attributes_payload(mso_values, protocols_dict)
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=interface_group_policy_path, value=mso_values))
+
+    elif state == "absent":
+        if mso.existing:
+            ops.append(dict(op="remove", path=interface_group_policy_path))
+
+    if not module.check_mode and ops:
+        mso_template.template = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        l3out_object = mso_template.get_l3out_object(l3out_uuid, l3out, True)
+        interface_groups = l3out_object.details.get("interfaceGroups", [])
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            interface_groups,
+            [KVPair("name", name)],
+        )
+        if match:
+            for protocol_name in protocols_dict.keys():
+                if match.details.get(protocol_name, {}).get("key", {}).get("ref"):
+                    match.details[protocol_name]["key"].pop("ref", None)
+            mso_template.update_config_with_template_and_references(match.details, reference_dict)
+            mso.existing = match.details
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        mso_template.update_config_with_template_and_references(mso.proposed, reference_dict)
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def check_authentication_requirements(mso, protocols_dict):
+    failed_attributes = {}
```

Comment:
not sure I understand the dict choice here? why not a list or directly a string, when keys are not used?